src/pg/sql/getArgSql.js

```diff
@@ -38,9 +38,11 @@ export default function getArgSql(
 
   let key;
   const where = [];
-  if (!isEmpty(filter)) {
-    where.push(getFilterSql(filter, options));
-    key = sql`${JSON.stringify(filter)}::jsonb`;
+  if (!isEmpty(filter) || $group === true) {
+    if (!isEmpty(filter)) {
+      where.push(getFilterSql(filter, options));
+    }
+    key = sql`${JSON.stringify({ ...filter, $group })}::jsonb`;
```

Comment:
:-1: On further thought, this variable should not contain the $group. The $group part of the $key is added later, in getAggMeta() so we should not add it here.

I suggest renaming this to:
```
const filterKey = ...
```
to make the distinction clearer.

Then further down, we can replace:
```
  key = sql`(${join([key, orderKey, cursorKey].filter(Boolean), ` || `)})`;
```
with
```
const   key = sql`(${join([filterKey, orderKey, cursorKey].filter(Boolean), ` || `)})`;
```